storage_devices/raid.py

```
@@ -0,0 +1,181 @@
+                    raid["uuid"],
+                    raid["container"]["uuid"] if "container" in raid else None,
+                    raid["container"]["path"] if "container" in raid else None,
+                    MetadataVariant(raid["metadata"]),
+                    [Device(d) for d in raid["array_devices"]],
+                    [Device(d) for d in raid["devices"]]
+                )
+            )
+
+        return raids
+
+    @classmethod
+    def create(
+            cls,
+            raid_configuration: RaidConfiguration,
+            devices: [Device]
+    ):
+        import copy
+        raid_conf = copy.deepcopy(raid_configuration)
+
+        if not raid_conf.number_of_devices:
+            raid_conf.number_of_devices = len(devices)
+        elif len(devices) < raid_conf.number_of_devices:
+            raise ValueError("RAID configuration requires at least "
+                             f"{raid_conf.number_of_devices} devices")
+
+        md_dir_path = "/dev/md/"
+
+        array_devices = devices
+        volume_devices = devices[:raid_conf.number_of_devices]
+
+        if raid_conf.metadata != MetadataVariant.Legacy:
+            container_conf = RaidConfiguration(
+                name=cls.__get_unique_name(raid_conf.metadata.value),
+                metadata=raid_conf.metadata,
+                number_of_devices=len(array_devices)
+            )
+            container_output = Mdadm.create(container_conf, get_devices_paths_string(array_devices))
+            if container_output.exit_code != 0:
+                raise CmdException(f"Failed to start {raid_conf.metadata.value} container",
+                                   container_output)
+
+        if not raid_conf.name:
+            raid_conf.name = cls.__get_unique_name()
+
+        raid_output = Mdadm.create(raid_conf, get_devices_paths_string(volume_devices))
+        if raid_output.exit_code != 0:
+            raise CmdException(f"Failed to start RAID volume", raid_output)
```

We could raise this exception inside `Mdadm.create()` instead of returning exit code.